config/federation/prometheus/prometheus.yml

```diff
@@ -373,6 +373,42 @@ scrape_configs:
         target_label: __address__
         replacement: blackbox-public-service.default.svc.cluster.local:9115
 
+  # SNMP configurations.
+  - job_name: 'snmp-targets'
+    metrics_path: /snmp
+
+    file_sd_configs:
+      - files:
+          - /snmp-targets/*.json
+        # Attempt to re-read files every five minutes.
+        refresh_interval: 5m
+
+    # This relabel config is necessary. The relabel config redefines the address
+    # to scrape and sets the correct parameters to pass to the scrape target.
+    relabel_configs:
+
+      # The default __address__ value is a target host from the config file.
+      # Here, we set (i.e. "replace") a request parameter "target" equal to the
+      # host value.
+      - source_labels: [__address__]
+        regex: (.*)(:9116)?
```

Comment:
Does this work in practice? My expectation is that it actually strips off the port suffix. The port 80 pattern from the blackbox exporter is b/c prometheus helpfully adds `:80` to target addresses that have no port already.

Reply:
It works, I think, because the target generated by mlabconfig.py is currently using a template like [s1.{{site}}.measurement-lab.org:9116](https://github.com/m-lab/operator/blob/master/generate_prometheus_targets.sh#L47). But maybe the target shouldn't have the port added, and I can simply add it here?